llama_index/vector_stores/postgres.py

```diff
@@ -331,15 +331,25 @@ def _apply_filters_and_limit(
     ) -> Any:
         import sqlalchemy
 
+        sqlalchemy_conditions = {
+            "or": sqlalchemy.sql.or_,
+            "and": sqlalchemy.sql.and_,
+        }
+
         if metadata_filters:
-            for filter_ in metadata_filters.legacy_filters():
-                bind_parameter = f"value_{filter_.key}"
-                stmt = stmt.where(  # type: ignore
-                    sqlalchemy.text(f"metadata_->>'{filter_.key}' = :{bind_parameter}")
+            if metadata_filters.condition not in sqlalchemy_conditions:
+                raise ValueError(
+                    f"Invalid condition: {metadata_filters.condition}. "
+                    f"Must be one of {list(sqlalchemy_conditions.keys())}"
                 )
-                stmt = stmt.params(  # type: ignore
-                    **{bind_parameter: str(filter_.value)}
+            stmt = stmt.where(
+                sqlalchemy_conditions[metadata_filters.condition](
+                    *(sqlalchemy.text(
+                        f"metadata_->>'{filter_.key}' = '{filter_.value}'"
+                    )
+                    for filter_ in metadata_filters.legacy_filters())
```

Comment:
@juleskuehn this is great.
btw `legacy_filters` is going to be deprecated. Can you change it to use `MetadataFilter` and `MetadataFilter` has operator support as well.
Related code : https://github.com/run-llama/llama_index/blob/main/llama_index/vector_stores/types.py#L82-L94

Reply:
Good idea. I'll give it a try.